pennylane/beta/interfaces/autograd.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the mixin interface class for creating differentiable quantum tapes with
+Autograd.
+"""
+# pylint: disable=protected-access
+import autograd.extend
+import autograd.builtins
+
+from pennylane import numpy as np
+
+from pennylane.beta.queuing import AnnotatedQueue
+
+
+class AutogradInterface(AnnotatedQueue):

```

Comment:
Yep! Exactly.

This is because the interface might be applied to a _subclass_ of the quantum tape --- we don't know what the inheritance structure of the class we're wrapping looks like, so easier to make the interface independent of the hierarchy.